lib/vsc/utils/timestamp.py

```diff
@@ -131,3 +131,22 @@ def write_timestamp(filename, timestamp):
     cache = FileCache(filename)
     cache.update('timestamp', timestamp_, 0)
     cache.close()
+
+
+def retrieve_timestamp_with_default(filename, start_timestamp=None, default_timestamp="2014010100000Z", delta=10):
+    """
+    Return a tuple consisting of the following values:
+    - the timestamp from the given file if the start_timestamp is not provided 
+      and fall back on the default if needed.
+    - the current time based on the given delta, defaulting to 10s.
+    """
+    timestamp = start_timestamp
+    if not start_timestamp:
+        timestamp = convert_to_unix_timestamp(read_timestamp(filename))
+
+    timestamp = timestamp or default_timestamp
+
+    logging.info("Using timestamp %s", start_timestamp)
+
+    current_time = datetime.datetime.now(tz=utc) + datetime.timedelta(seconds=delta)
+    return (timestamp, current_time)
```

Comment:
i would also add a function to do
```
            (_, ldap_timestamp) = convert_timestamp(start_time)
            write_timestamp(SYNC_TIMESTAMP_FILENAME, ldap_timestamp)
```
or support new option to write_timestamp to do the conversion

Reply:
The question then really becomes if we still need this, no? If we'd store unix epoch everywhere we'd be fine, no?

Reply:
again, different PR as that would affect multiple repos
i don't mind stporing it n multiple formats, but the value should be correct ;)

the 2nd value is the one that is used and is human readable, so i'd say non-trivial change

Reply:
Yes, that would be my main concern.

Reply:
OK, I'll keep the above for another PR then.